relay-metrics/src/aggregation.rs

```diff
@@ -1742,36 +1864,12 @@ impl Aggregator {
         partitions
     }
 
-    /// Split the provided buckets into batches and process each batch with the given function.
-    ///
-    /// For each batch, log a histogram metric.
-    fn process_batches<F>(&self, buckets: impl IntoIterator<Item = Bucket>, mut process: F)
-    where
-        F: FnMut(Vec<Bucket>),
-    {
-        let capped_batches =
-            CappedBucketIter::new(buckets.into_iter(), self.config.max_flush_bytes);
-
-        let num_batches = capped_batches
-            .map(|batch| {
-                relay_statsd::metric!(
-                    histogram(MetricHistograms::BucketsPerBatch) = batch.len() as f64,
-                );
-                process(batch);
-            })
-            .count();
-
-        relay_statsd::metric!(
-            histogram(MetricHistograms::BatchesPerPartition) = num_batches as f64,
-        );
-    }
-
     /// Sends the [`FlushBuckets`] message to the receiver in the fire and forget fashion. It is up
     /// to the receiver to send the [`MergeBuckets`] message back if buckets could not be flushed
     /// and we require another re-try.
     ///
     /// If `force` is true, flush all buckets unconditionally and do not attempt to merge back.
-    fn try_flush(&mut self) {
+    async fn try_flush(&mut self, sender: CompatSender) {
```

Comment:
Instead of passing the `sender` as parameter, can you access `self.recipient` and use `compat::send_to_recipient` directly? Particularly, please also note my comment below about `do_send`.